x/fundraising/types/utils.go

```diff
@@ -34,8 +34,98 @@ func DeriveAddress(addressType AddressType, moduleName, name string) sdk.AccAddr
 
 // SortByBidPrice sorts bid array by bid price in descending order.
 func SortByBidPrice(bids []Bid) []Bid {
-	sort.SliceStable(bids, func(i, j int) bool {
-		return bids[i].Price.GT(bids[j].Price)
+	sort.Slice(bids, func(i, j int) bool {
+		if bids[i].Price.GT(bids[j].Price) {
+			return true
+		}
+		return bids[i].Id < bids[j].Id
 	})
 	return bids
 }
+
+func BidsByPrice(bids []Bid) (prices []sdk.Dec, bidsByPrice map[string][]Bid) {
+	bids = SortByBidPrice(bids)
+
+	bidsByPrice = map[string][]Bid{} // price => []Bid
+
+	for _, bid := range bids {
+		priceStr := bid.Price.String()
+		bidsByPrice[priceStr] = append(bidsByPrice[priceStr], bid)
+	}
+
+	// Sort prices in descending order.
+	prices = make([]sdk.Dec, len(bidsByPrice))
+	i := 0 // TODO: is it too much optimization? we can use append(...)
```

Comment:
It is good to know that simple assignment has better performance than appending. I think it is good as is.